extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/WindowOperatorQueryKit.java

```diff
@@ -112,7 +111,24 @@ public QueryDefinition makeQueryDefinition(
         false
     );
 
-    dataSourcePlan.getSubQueryDefBuilder().ifPresent(queryDefBuilder::addAll);
+    final QueryDefinitionBuilder builder = QueryDefinition.builder(queryId);
```

Comment:
nit: why its better to have 2 live `QueryDefinitionBuilder` typed varaibles in the scope instead of just 1; (why not drop final)?

Reply:
Made the change, now we have only 1 QueryDefinitionBuilder :)